### DISCUSSION TITLE
Routing - external routing integration

### DISCUSSION THREAD

**Discussion by Brian-McBride** [chrisdmacrae/astro-pwa#4] [Ideas]
I'd love to see the project with wrappers around routers like [Tanstack Router](https://tanstack.com/router/v1), [React Router](https://reactrouter.com/en/main), etc.

There is so much tooling and features in these.

I have a hybrid need where much of my site I want in something like Astro, but I have an actual app need. SPA/PWA. I think at this point I might just stick with CRA or similar and run that in `app.mydomain.com`.  But I sure would rater have `mydomain.com/app` 

What is the main challenge with routing? It seems that if I took a subset `src/pages/app/[...slug].astro` and then on my server I did a SPA like redirect where everything in that app route runs into `/app` - wouldn't I just end up with a SPA in Astro? Dropping something like [Tanstack Router](https://tanstack.com/router/v1) into that one route, I would think would capture routing to be client-side. I'd want to break out of it for routes external to that app path of course.

**Comment by chrisdmacrae** [comment]
@Brian-McBride it's not documented, but this tool generates a full client router with a public API from the filesystem routes you glob.

There's nothing stopping us from adapting the routes in that router to TanStack router or React Router, honestly.

**Comment by chrisdmacrae** [comment]
https://github.com/chrisdmacrae/astro-pwa/blob/main/packages/astro-pwa/src/routing/router.ts

```
import { useRouter } from 'astro-pwa/react'

export const MyCompontentInsideSPA = () => {
   const router = useRouter()
   const routes = router.routes
}
```

**Reply by chrisdmacrae** [reply]
@Brian-McBride to add to this.

I want the DX of services like Next/Nuxt/Remix etc where filesystem routes are server-side rendered and then hydrated.

I don't want a SPA that doesn't exist on the server.
